src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/ConfigUpdater.cs

```diff
@@ -0,0 +1,184 @@
+    {
+        private readonly XDocument _config;
+        private readonly ILogger<ConfigUpdater> _logger;
+
+        public ConfigUpdater(XDocument doc, ILogger<ConfigUpdater> logger)
+        {
+            _config = doc;
+            _logger = logger;
+        }
+
+        // Updates the original config file by removing the system.serviceModel element
+        public XDocument UpdateOldConfig()
+        {
+            var oldConfig = new XDocument(_config);
+            var serviceModel = oldConfig.Root.DescendantsAndSelf("system.serviceModel");
+            serviceModel.Single().AddBeforeSelf(new XComment(Constants.ServiceModelComment));
+            serviceModel.Remove();
+
+            _logger.LogDebug("The original config file finished updating. System.serviceModel element was removed.");
+            return oldConfig;
+        }
+
+        public XDocument GenerateNewConfig()
+        {
+            var wcfConfig = new XDocument(new XElement("configuration", _config.Root.DescendantsAndSelf("system.serviceModel")));
+
+            // comment out host and behavior elements which are not supported by CoreWCF and configured in the code instead
+            var baseAddress = wcfConfig.Root.DescendantsAndSelf("host");
+            var serviceBehavior = wcfConfig.Root.DescendantsAndSelf("behaviors");
+            if (baseAddress.Any())
+            {
+                var host = baseAddress.First();
+                host.AddBeforeSelf(new XComment(Constants.HostComment));
+                host.ReplaceWith(new XComment(host.ToString()));
+            }
+
+            if (serviceBehavior.Any())
+            {
+                var behaviors = serviceBehavior.Single();
+                behaviors.AddBeforeSelf(new XComment(Constants.BehaviorComment));
+                behaviors.ReplaceWith(new XComment(behaviors.ToString()));
+            }
+
+            // find and remove the mex endpoint
+            if (IncludesMexEndpoint())
+            {
+                var unsupported_endpoint = from el in wcfConfig.Root.DescendantsAndSelf("endpoint")
+                                           where el.Attribute("binding").Value.StartsWith("mex", StringComparison.Ordinal)
+                                           select el;
+                unsupported_endpoint.First().AddBeforeSelf(new XComment(Constants.MexEndpoint));
+                unsupported_endpoint.Remove();
+                _logger.LogWarning("The mex endpoint is removed from .config and service metadata behavior is configured in the source code instead.");
+            }
+
+            wcfConfig = UpdateEndpoints(wcfConfig);
+            _logger.LogDebug("Finished creating the new configuration file.");
+            return wcfConfig;
+        }
+
+        // Adds path from base address to the beginning of endpoints if needed
+        private XDocument UpdateEndpoints(XDocument config)
+        {
+            // get base address
+            var uri = GetUri();
+
```

Comment:
It's not clear what the return type is here. Either rename the method to be clearer what's returned, or explicitly specify the data type for uri. Maybe GetSchemeToBasepathMapping if renaming?